exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpGrpcSpanExporterComponentProvider.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.internal;
+
+import static io.opentelemetry.exporter.otlp.internal.OtlpConfigUtil.DATA_TYPE_TRACES;
```

Comment:
this and the next classes all look very similar - can we make this more concise?

Reply:
I _think_ this is what @zeitlinger was kinda getting at: https://github.com/jack-berg/opentelemetry-java/pull/19 feel free to ignore it, but I took a stab at seeing what it might be like to start some consolidation there.

Reply:
Yup I get the idea. Couple of thoughts:

1. Let's consider this separate from this PR.
2. We already have a similar concept in [TelemetryExporterBuilder](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/TelemetryExporterBuilder.java). This is a test utility, but is an example of where it is beneficial of to have a common base interface for the builder. But what's not clear to me is how end users would benefit from this. So far, I've only seen internal code stand to benefit from introduction.

Reply:
If you can think of a user facing benefit, please open an issue!